pkg/resources/provisioning.cattle.io/v1/cluster/validator.go

```diff
@@ -113,6 +126,108 @@ func (p *provisioningAdmitter) Admit(request *admission.Request) (*admissionv1.A
 	return response, nil
 }
 
+// validateAgentEnvVars will validate the provisioning cluster object's AgentEnvVars, ensuring that the
+// "CATTLE_AGENT_VAR_DIR" env var is not allowed to be updated.
+func (p *provisioningAdmitter) validateAgentEnvVars(request *admission.Request, oldCluster, newCluster *v1.Cluster) *metav1.Status {
+	if newCluster.Spec.RKEConfig == nil {
+		return nil
+	}
+	if request.Operation == admissionv1.Create {
+		if slices.ContainsFunc(newCluster.Spec.AgentEnvVars, func(envVar rkev1.EnvVar) bool {
+			return envVar.Name == "CATTLE_AGENT_VAR_DIR"
```

Comment:
This env var sees a lot of re-use across this function and the related function. I think it would be a good idea to set it as a const at the top of the file to avoid typos.